CASSGO-81 Add missing Context variant methods

joao-r-reis opened the pull request:
In #1868  we deprecated Query.WithContext and Batch.WithContext. We added ExecContext and IterContext as alternatives but we didn't add these alternatives for other ways to execute queries and batches like Query.Scan or Batch.ExecCAS.

We also didn't deprecate Query.Context() and we should have

joao-r-reis commented:
~~TODO: add to changelog and rewrite commit msg~~

jameshartig commented on session.go:
```diff
@@ -786,7 +786,16 @@ func (s *Session) ExecuteBatchCAS(batch *Batch, dest ...interface{}) (applied bo
 // the applied boolean as the first argument to *Iter.Scan
 func (b *Batch) ExecCAS(dest ...interface{}) (applied bool, iter *Iter, err error) {
-	iter = b.session.executeBatch(b, nil)
+	return b.ExecCASContext(nil, dest...)
```
The context docs say:
> Do not pass a nil [Context](https://pkg.go.dev/context#Context), even if a function permits it. Pass [context.TODO](https://pkg.go.dev/context#TODO) if you are unsure about which Context to use.

We should either pass a TODO or pass context.Background()

joao-r-reis replied:
yeah I thought about that but then the internal code that defaults to the old context field that is deprecated will not work...

joao-r-reis replied:
maybe I can use context.TODO and treat context.TODO as if the user didn't provide it

jameshartig replied:
Oh I didn't make the connection with `newQueryOptions` and it checking for nil specifically. Could we just do `q.Context()` in the query ones and `b.Context()` in the batch ones here instead off `nil`?

joao-r-reis replied:
oh I didn't think about that, yeah I'll do that

joao-r-reis committed:
Add missing Context variant methods to Query and Batch

In CASSGO-22 we deprecated WithContext but we didn't deprecate Context().
IterContext and ExecContext were added as replacements but there are
other execution methods that lack a Context variant.

This patch addresses both of these issues.

Patch by João Reis; reviewed by James Hartig for CASSGO-81
